scripts/monitor_submission.py

```diff
@@ -49,14 +49,11 @@ def monitor_submission(terra_workspace, terra_project, submission_id, sleep_time
             # terra_details to be appended to json
             terra_details = {"terra_workspace": terra_workspace, "terra_project": terra_project}
 
-            # loading JSON submission_metadata:
-            submission_data = json.loads(submission_metadata)
-
-            # appending the terra_details
-            submission_data.update(terra_details)
+            # udating/appending the terra_details to current json
+            submission_metadata.update(terra_details)
 
```

Comment:
you could move the `with open(save_name, 'w') as f:` line to here -- the updating of submission_metadata with terra_details doesn't need to be inside that save block.